include/wil/registry.h

```diff
@@ -17,16 +17,2704 @@
 
 #include <winreg.h>
 #include <new.h> // new(std::nothrow)
-#include "resource.h" // unique_hkey
+#include "registry_helpers.h"
+#include "resource.h"
+
+#ifndef __WIL_WINREG_
+#error __WIL_WINREG_ is required for ::wil::unique_hkey support; ensure the raw registry APIs (like RegCreateKeyExA) are available
+#endif
+
+// wil registry does not require the use of the STL or C++ exceptions (see _nothrow functions)
+// wil registry natively supports std::vector and std::wstring when preferring those types
+// wil registry uses the __WIL_WINREG_STL define to enable support for wil::shared_* types (defined in resource.h)
 
 namespace wil
 {
-    //! The key name includes the absolute path of the key in the registry, always starting at a
-    //! base key, for example, HKEY_LOCAL_MACHINE.
-    size_t const max_registry_key_name_length = 255;
+    namespace reg
+    {
+        /**
+         * \brief Helper function to translate registry return values if the value was not found
+         * \param hr HRESULT to test from registry APIs
+         * \return boolean if the HRESULT indicates the registry value was not found
+         */
+        constexpr bool is_hresult_not_found(HRESULT hr) WI_NOEXCEPT

```

Comment:
given `ERROR_NOT_FOUND` exists, and this is not checking for that... rename to `is_registry_not_found` or `is_file_not_found`
```cpp
#define ERROR_NOT_FOUND                  1168L
#define ERROR_FILE_NOT_FOUND             2L
```

Reply:
Thanks!